app/build.gradle

```diff
@@ -143,30 +143,47 @@ if (project.hasProperty("telemetry") && project.property("telemetry") == "true")
     if (variant.buildType.buildConfigFields['IS_RELEASED']?.value) {
         // The Google Play Store does not allow multiple APKs for the same app that all have the
         // same version code. Therefore we need to have different version codes for our ARM and x86
-        // builds.
+        // builds. See https://developer.android.com/studio/publish/versioning
 
         // Our x86 builds need a higher version code to avoid installing ARM builds on an x86 device
         // with ARM compatibility mode.
 
+        // AAB builds need a version code that is distinct from any APK builds. Since AAB and APK
+        // builds may run in parallel, AAB and APK version codes might be based on the same
+        // (minute granularity) time of day. To avoid conflicts, we ensure the minute portion
+        // of the version code is even for APKs and odd for AABs.
+
         def versionName = Config.releaseVersionName(project)
 
         variant.outputs.each { output ->
             def abi = output.getFilter(FilterConfiguration.FilterType.ABI.name())
+            def aab = project.hasProperty("aab")
+
+            // ensure baseVersionCode is an even number
+            if (baseVersionCode % 2) {
+                baseVersionCode = baseVersionCode + 1
+            }
 
             def versionCodeOverride = baseVersionCode
 
-            if (abi == "x86_64") {
-                versionCodeOverride = baseVersionCode + 3
-            } else if (abi == "x86") {
-                versionCodeOverride = baseVersionCode + 2
-            } else if (abi == "arm64-v8a") {
+            if (aab) {
+                // AAB version code is odd
                 versionCodeOverride = baseVersionCode + 1
```

Comment:
Question: should the AAB have the greatest `versionCode`? This way, users default to the regular APKs only when they can't get the ones generated by Google?

Reply:
I think it will be difficult to ensure that the AAB versionCode is greater than that of any APKs:
 - generating a base time in the decision task and passing that through to build tasks seems fragile and error prone
 - building APKs and AABs serially in one task would increase build durations.
As discussed, I'm not sure we need that: We will normally upload either an AAB or a set of APKs, but not both, from one graph. Play should not allow both formats on the same track, regardless of version code. Recent experience on reference-browser shows that Play does not reject the upload of an APK with a smaller version code than the version code of an AAB already uploaded to a different track.

Reply:
> * generating a base time in the decision task and passing that through to build tasks seems fragile and error prone

True! I had forgotten that. Let's go with the this implementation then 👍